### ISSUE TITLE
nans with bounded uniform priors in version 0.51.0 

### ISSUE THREAD

**Issue by beanSnippet** [libqueso/queso#316]
I have 15 parameters, and I'm trying to use a bounded uniform prior for each. When I use v51, I get this output:
```
WARNING In MetropolisHastingsSG<P_V,P_M>::alpha(x,y), worldRank 0, fullRank 0, subEnvironment 0, subRank 0, inter0Rank 0, positionId = 4, stageId = 0: y.logTarget() = nan, x.values() = 5.99997 3.1455 1.09257e-09 5.9552 1.74279 -0.311433 0.59255 0.28331 -2.48333 -0.485159 100 100 1 10.3607 1 , y.values() = 6 4.81996 9.39886e-12 5.99406 1.42883 -0.265608 0.397198 0.223693 -2.41517 -1.0726 100 -nan 1 19.9992 1
```
That might be hard to read. The main point is that `y.logTarget() = nan`, and one (or more) of y.values is nan.

I don't see this with v50.

**Comment by dmcdougall** [comment]
Try one of the following (not both):

1. Set `ip_mh_doLogitTransform = 0` in the queso input file;
2. Decrease the proposal covariance matrix (by several orders of magnitude).

Do either of these help?

**Comment by beanSnippet** [comment]
#1 seems to fix it. I haven't tried #2 yet, although I was messing around the covariance earlier and couldn't get anything to work. Is it useful for you to know if that helps?

**Comment by dmcdougall** [comment]
>#1 seems to fix it

Awesome.

>I was messing around the covariance earlier and couldn't get anything to work. Is it useful for you to know if that helps?

Yes.  If it helps, then it tells me the heuristic that maps the proposal covariance matrix to the transformed space is not good enough, and I can release a bugfix version of queso that makes the logit transform logic off by default.

**Comment by beanSnippet** [comment]
So I made the covariance really small, and I didn't get any nans if there was no adaption (`ip_mh_am_initialNonAdaptInterval` > `ip_mh_rawChain_size`). Otherwise, starts to get nans after the adaption kicks in.

**Comment by csim314** [comment]
Sounds like a regression test should come of this.

**Comment by dmcdougall** [comment]
I need the following information on parameter 12:

1.  The size of the domain;
2.  Initial value passed to QUESO for the Markov chain;
3.  Assuming your proposal covariance is diagonal, the associated proposal variance.

**Comment by beanSnippet** [comment]
So, I didn't realize you might need this info, and my code has changed since then. I just tried to recreate the problem, but it didn't happen immediately. I think the info for parameter 12 was something like:

1. domain: (1,2)
2. initial position: 1.5
3. proposal covariance on the diagonal: 0.075

I will let you know the details again if I can recreate it.

**Comment by dmcdougall** [comment]
@beanSnippet  Would you mind trying out my `fix-logit-proposal-covariance` branch?  First without adaptivity and then with adaptivity.  Do you still get `NaN`s?

**Comment by beanSnippet** [comment]
With my code as I have the domain and proposal set now, it looks good, with
and without adaptivity.

However, I can force nans to happen if I give the bounded domain a too
large proposal (like, 10 on the diagonal element for a domain of size 1). I
think there used to be an error message for this (?), asking the user to
decrease the initial proposal for the given domain.

**Comment by dmcdougall** [comment]
Closing this since it seems to have been resolved with #337.  Feel free to open a new ticket if the issue returns.
